src/main/java/migrator/app/domain/index/IndexRepository.java

```diff
@@ -0,0 +1,74 @@
+package migrator.app.domain.index;
+
+import java.util.Collection;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import migrator.app.domain.table.model.Index;
+import migrator.app.migration.model.ChangeCommand;
+import migrator.app.migration.model.IndexProperty;
+import migrator.lib.repository.UniqueRepository;
+
+public class IndexRepository extends UniqueRepository<Index> {
+    protected UniqueRepository<IndexProperty> indexPropertyRepo;
+    protected UniqueRepository<ChangeCommand> changeCommandRepo;
+    protected Map<String, ObservableList<Index>> byTable;
+
+    public IndexRepository(UniqueRepository<IndexProperty> indexPropertyRepo, UniqueRepository<ChangeCommand> changeCommandRepo) {
+        super();
+        this.changeCommandRepo = changeCommandRepo;
+        this.indexPropertyRepo = indexPropertyRepo;
+        this.byTable = new Hashtable<>();
```

Comment:
Create RepositoryMapIndex implements Repository<T> that will add items by some key into array. So you can index by tableId for example